src/app/loginComponents/accounts/accounts.component.ts

```diff
@@ -1,12 +1,53 @@
 import { Component, OnInit } from '@angular/core';
-import { Subject } from 'rxjs';
+import { Location } from '@angular/common';
+import { MatTabChangeEvent } from '@angular/material';
+import { FormControl } from '@angular/forms';
+import { Router } from '@angular/router';
 
 @Component({
   selector: 'app-accounts',
   templateUrl: './accounts.component.html'
 })
 export class AccountsComponent implements OnInit {
+  public currentTab = 'accounts'; // default to the 'accounts' tab
+  protected selected = new FormControl(0);
+  validTabs = ['accounts', 'profiles', 'dockstore account controls', 'requests'];
+  constructor(private location: Location, private router: Router) {}
+
   ngOnInit() {
     localStorage.setItem('page', '/accounts');
+    this.parseURL(this.router.url);
+  }
+
+  private parseURL(url: String): void {
+    const decodedUrl: string = decodeURIComponent(url.toString()); // decode any encoded spaces, etc. in the string
+    const strippedUrl: String = decodedUrl.endsWith('/') ? decodedUrl.slice(0, -1) : decodedUrl; // remove any trailing slash
+    this.setupTab(strippedUrl);
+  }
+
+  public setupTab(url: String) {
+    const regexp: RegExp = new RegExp('\\?tab=(.*)');
+    const match = regexp.exec(url.toString());
+    if (match) {
+      // look for a tab name in the url
+      this.currentTab = match[1]; // if found, set the tab accordingly
+    } // if not found, default to the 'accounts' tab
+    this.selectTab(this.validTabs.indexOf(this.currentTab));
```

Comment:
What happens if the URL contains an invalid tab name? Normally it shouldn't happen, but somebody could type in the url and have a typo. You should probably catch that.

Reply:
I believe it defaults to the accounts tab when that happens, as if you hadn't specified a tab

Reply:
Could be worth adding a test that navigates to a non-existent tab and verifies that it behaves like that.

Reply:
When the url has an invalid tab it goes to the accounts page, but does not update the url to point to accounts. I think that would be nice to have working.